## Baseline predictions of Human Development and State Failure events are meant to emulate a human making an educated guess simply based on a nation-states performance in the most recent years, according to this dataset.

In [1]:
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.metrics import f1_score

In [2]:
df = pd.read_excel('Master_MELT.xlsx')
df

NS_code_from_gini Nation_State  T_HDI_Mean_plus_5  T_HDI_Trend_plus_5  \
0                  AFG  Afghanistan           0.305000            0.001400   
1                  ALB      Albania           0.621000           -0.006800   
2                  DZA      Algeria           0.587800            0.004900   
3                  AGO       Angola           0.317822            0.010456   
4                  ARG    Argentina           0.718800            0.005500   
...                ...          ...                ...                 ...   
5513               YEM  Yemen, Rep.                NaN                 NaN   
5514               ZMB       Zambia                NaN                 NaN   
5515               ZWE     Zimbabwe                NaN                 NaN   
5516               SOM      Somalia                NaN                 NaN   
5517               XKX       Kosova                NaN                 NaN   

      T_State_Fail_yr_1  T_State_Fail_plus_5  T_State_Fail_yr_5  year0  \
0                   1.0                  1.0                1.0   1989   
1                   0.0                  0.0                0.0   1989   
2                   0.0                  1.0                1.0   1989   
3                   1.0                  1.0                1.0   1989   
4                   0.0                  0.0                0.0   1989   
...                 ...                  ...                ...    ...   
5513                NaN                  NaN                NaN   2019   
5514                NaN                  NaN                NaN   2019   
5515                NaN                  NaN                NaN   2019   
5516                NaN                  NaN                NaN   2019   
5517                NaN                  NaN                NaN   2019   

      CWGI_Poli_Stabl_Mean_minus_4  CWGI_Poli_Stabl_Trend_minus_4  \
0                        -2.448409                       0.000183   
1                        -1.076163                       0.042262   
2                        -2.215933                       0.043574   
3                        -3.513859                       0.133881   
4                        -0.395055                       0.014940   
...                            ...                            ...   
5513                     -2.686827                      -0.078960   
5514                      0.650865                       0.039156   
5515                     -0.938675                       0.009522   
5516                     -3.195744                      -0.039098   
5517                     -1.959296                      -0.251318   

      CWGI_RoL_Mean_minus_4  CWGI_RoL_Trend_minus_4  E_GINI_Mean_minus_4  \
0                 -1.757795               -0.001501            26.457876   
1                 -1.598236                0.044494            28.631707   
2                 -1.312030                0.027104            40.234099   
3                 -2.008206                0.031328            52.547934   
4                  0.223199               -0.041860            44.060000   
...                     ...                     ...                  ...   
5513              -1.030670                0.017181            34.706441   
5514              -0.346509                0.011784            52.486854   
5515              -2.060016               -0.048014            53.805381   
5516              -2.553994               -0.017658            35.613447   
5517              -0.283984                0.052486            27.236929   

      E_GINI_Trend_minus_4  PCPI_std_Mean_minus_4  PCPI_std_Trend_minus_4  \
0                 0.193825               0.030527               -0.055970   
1                 0.068211              -1.365715                0.036453   
2                -0.332204              -0.654554                0.004671   
3                -0.306912              -0.939852               -0.007850   
4                 1.760000              -0.936350   

In [3]:
df.describe()

T_HDI_Mean_plus_5  T_HDI_Trend_plus_5  T_State_Fail_yr_1  \
count        4984.000000         4984.000000        5340.000000   
mean            0.654033            0.004692           0.128652   
std             0.168536            0.004015           0.334845   
min             0.168917           -0.028400           0.000000   
25%             0.523000            0.002312           0.000000   
50%             0.679800            0.004600           0.000000   
75%             0.786450            0.007006           0.000000   
max             0.953667            0.029200           1.000000   

       T_State_Fail_plus_5  T_State_Fail_yr_5        year0  \
count          4722.000000        4628.000000  5518.000000   
mean              0.222363           0.117329  2004.000000   
std               0.415878           0.321847     8.945082   
min               0.000000           0.000000  1989.000000   
25%               0.000000           0.000000  1996.000000   
50%               0.000000           0.000000  2004.000000   
75%               0.000000           0.000000  2012.000000   
max               1.000000           1.000000  2019.000000   

       CWGI_Poli_Stabl_Mean_minus_4  CWGI_Poli_Stabl_Trend_minus_4  \
count                   5518.000000                    5518.000000   
mean                      -0.117528                      -0.003093   
std                        1.041123                       0.087161   
min                       -3.567753                      -0.680213   
25%                       -0.787559                      -0.039650   
50%                       -0.018512                      -0.001824   
75%                        0.704468                       0.036635   
max                        5.580238                       0.397186   

       CWGI_RoL_Mean_minus_4  CWGI_RoL_Trend_minus_4  E_GINI_Mean_minus_4  \
count            5518.000000             5518.000000          5518.000000   
mean               -0.120058                0.001144            40.055082   
std                 1.009495                0.046029             9.632331   
min                -3.090691               -0.402811             5.103747   
25%                -0.871368               -0.022753            32.698167   
50%                -0.289734                0.003940            38.786059   
75%                 0.602019                0.025398            45.869545   
max                 2.022047                0.230820            78.454219   

       E_GINI_Trend_minus_4  PCPI_std_Mean_minus_4  PCPI_std_Trend_minus_4  \
count           5518.000000            5518.000000             5518.000000   
mean              -0.074963              -0.256538                0.022170   
std                0.942308               1.321091                0.147354   
min               -4.857500              -7.724480               -1.291164   
25%               -0.569167              -0.913885               -0.037947   
50%               -0.090000              -0.444033                0.018115   
75%                0.374769               0.441417                0.079808   
max                6.074167               5.180655                0.896241   

       PWGI_Corrupt_Ctrl_Mean_minus_4  PWGI_Corrupt_Ctrl_Trend_minus_4  \
count                     5518.000000                      5518.000000   
mean                        -0.064260                        -0.002037   
std                          1.021928                         0.054134   
min                         -2.334239                        -0.239165   
25%                         -0.820101                        -0.030105   
50%                         -0.321296                        -0.003605   
75%                          0.517726                         0.025411   
max                          2.655868                         0.321575   

       PWGI_Voice_Acct_Mean_minus_4  PWGI_Voice_Acct_Trend_minus_4  
count                   5518.000000                    5518.000000  
mean       

In [4]:
df.head(1)

NS_code_from_gini Nation_State  T_HDI_Mean_plus_5  T_HDI_Trend_plus_5  \
0               AFG  Afghanistan              0.305              0.0014   

   T_State_Fail_yr_1  T_State_Fail_plus_5  T_State_Fail_yr_5  year0  \
0                1.0                  1.0                1.0   1989   

   CWGI_Poli_Stabl_Mean_minus_4  CWGI_Poli_Stabl_Trend_minus_4  \
0                     -2.448409                       0.000183   

   CWGI_RoL_Mean_minus_4  CWGI_RoL_Trend_minus_4  E_GINI_Mean_minus_4  \
0              -1.757795               -0.001501            26.457876   

   E_GINI_Trend_minus_4  PCPI_std_Mean_minus_4  PCPI_std_Trend_minus_4  \
0              0.193825               0.030527                -0.05597   

   PWGI_Corrupt_Ctrl_Mean_minus_4  PWGI_Corrupt_Ctrl_Trend_minus_4  \
0                       -2.046923                         0.024895   

   PWGI_Voice_Acct_Mean_minus_4  PWGI_Voice_Acct_Trend_minus_4  
0                     -2.322069                       0.045336

## The dataframe created below aligns last year's state failure events with state failures predicted 5 years from now.

In [5]:
State_fail_yr_5 = df[df.year0 < 2014].iloc[:,4:5]
State_fail_yr_5.columns=['predicted']
ninety_19 = df[df.year0 > 1989]
ninety_14 = ninety_19[ninety_19['year0'] < 2015]
State_fail_yr_5['actual'] = np.array(ninety_14.iloc[:,6:7])
State_fail_yr_5

predicted  actual
0           1.0     1.0
1           0.0     0.0
2           0.0     1.0
3           1.0     0.0
4           0.0     0.0
...         ...     ...
4445        1.0     1.0
4446        0.0     0.0
4447        0.0     1.0
4448        1.0     0.0
4449        0.0     0.0

[4450 rows x 2 columns]

In [6]:
f1_score(State_fail_yr_5['actual'], State_fail_yr_5['predicted'])

0.4903581267217631

## BASELINE: T_State_Fail_yr_1 (state failure on year one), predicted by state failure in year 0, an F-Score of 0.80547.

## BASELINE: T_State_Fail_yr_5 (state failure on year five), predicted by state failure in year 0, an F-Score of 0.49036.

In [8]:
State_fail_nxt_5 = df[df.year0 < 2014].iloc[:,4:5]
State_fail_nxt_5.columns=['predicted']
ninety_19 = df[df.year0 > 1989]
ninety_14 = ninety_19[ninety_19['year0'] < 2015]
State_fail_nxt_5['actual'] = np.array(ninety_14.iloc[:,5:6])
State_fail_nxt_5

predicted  actual
0           1.0     1.0
1           0.0     0.0
2           0.0     1.0
3           1.0     1.0
4           0.0     0.0
...         ...     ...
4445        1.0     1.0
4446        0.0     0.0
4447        0.0     1.0
4448        1.0     1.0
4449        0.0     0.0

[4450 rows x 2 columns]

## The dataframe above aligns last year's state failure events with state failures predicted in any of the next 5 years.

In [9]:
f1_score(State_fail_nxt_5['actual'], State_fail_nxt_5['predicted'])

0.6684709066305818

## ^^ BASELINE: T_State_Fail_plus_5 (state failure in any of the next 5 years), predicted by state failure in year 0, an F-Score of 0.6685 ^^

In [11]:
HDI_Mean = df[df.year0 < 2012].iloc[:,2:3]
HDI_Mean.columns=['predicted']

eighty9_16 = df[df.year0 < 2017]
ninety4_16 = eighty9_16[eighty9_16.year0 > 1993]

HDI_Mean['actual'] = np.array(ninety4_16.iloc[:,2:3])
HDI_Mean

predicted    actual
0      0.305000  0.335000
1      0.621000  0.643200
2      0.587800  0.619600
3      0.317822  0.368699
4      0.718800  0.746000
...         ...       ...
4089   0.496200  0.463000
4090   0.565200  0.590333
4091   0.534600  0.559667
4092   0.311400  0.350831
4093   0.745100  0.744361

[4094 rows x 2 columns]

In [12]:
from sklearn.metrics import mean_squared_error
from sklearn.metrics import r2_score

In [13]:
(mean_squared_error(HDI_Mean['actual'], HDI_Mean['predicted']))**0.5

0.030119237168458646

In [14]:
r2_score(HDI_Mean['actual'], HDI_Mean['predicted'])

0.9668353998262852

## ^^ BASELINE: T_HDI_Mean_plus_5 (HDI Mean in the next 5 years), predicted by HDI Mean in the last 5 years, an RMSE of 0.03012, and an r2_score of 0.9668354. ^^

In [16]:
HDI_Trend = df[df.year0 < 2012].iloc[:,3:4]
HDI_Trend.columns=['predicted']

eighty9_16 = df[df.year0 < 2017]
ninety4_16 = eighty9_16[eighty9_16.year0 > 1993]

HDI_Trend['actual'] = np.array(ninety4_16.iloc[:,3:4])
HDI_Trend

predicted    actual
0      0.001400  0.004000
1     -0.006800  0.007200
2      0.004900  0.009100
3      0.010456  0.009053
4      0.005500  0.007800
...         ...       ...
4089  -0.006100  0.000000
4090   0.006700  0.001000
4091   0.008300  0.005000
4092   0.013193  0.001487
4093  -0.000300  0.002095

[4094 rows x 2 columns]

In [17]:
r2_score(HDI_Trend['actual'], HDI_Trend['predicted'])

-0.6273702311931042

In [18]:
(mean_squared_error(HDI_Trend['actual'], HDI_Trend['predicted']))**0.5

0.004824537565757067

## ^^ BASELINE: T_HDI_Trend_plus_5 (HDI Trend in the next 5 years), predicted by HDI Trend in the last 5 years, an RMSE of 0.004824, and an r2_score of -0.62737. ^^